cfgrammar/src/lib/yacc/ast.rs

```diff
@@ -202,8 +230,101 @@ impl GrammarAST {
                 spans: vec![Span::new(0, 0)],
             });
         }
+
+        for sym in &self.expect_unused {
+            match sym {
+                Symbol::Rule(sym_name, sym_span) => {
+                    if self.get_rule(sym_name).is_none() {
+                        return Err(YaccGrammarError {
+                            kind: YaccGrammarErrorKind::UnknownRuleRef(sym_name.clone()),
+                            spans: vec![*sym_span],
+                        });
+                    }
+                }
+                Symbol::Token(sym_name, sym_span) => {
+                    if !self.has_token(sym_name) {
+                        return Err(YaccGrammarError {
+                            kind: YaccGrammarErrorKind::UnknownToken(sym_name.clone()),
+                            spans: vec![*sym_span],
+                        });
+                    }
+                }
+            }
+        }
         Ok(())
     }
+
+    /// Returns tokens, and rules unreachable from the start rule
+    /// by index into the `ast.tokens` and `ast.rules` fields.
```

Comment:
I can parse line 257 but not line 258. I *think* it means "Return the indices of unexpectedely unused rules (relative to `ast.rules`) and tokens (relative to `ast.tokens`)."?

Reply:
Yes, exactly I should probably add "as `SymbolIdx`s"